### PR TITLE
fix: use latest subsidy expiration date for universal links, misc fixes

### PR DESCRIPTION
**Pull request by long74100**
# For all changes

- [x] Ensure adequate tests are in place (or reviewed existing tests cover changes)

# Only if submitting a visual change

- [ ] Ensure to attach screenshots
- [ ] Ensure to have UX team confirm screenshots

### CODE REVIEW

File: src/components/settings/SettingsAccessTab/SettingsAccessLinkManagement.jsx [hunk 1]
```diff
@@ -42,6 +44,12 @@ const SettingsAccessLinkManagement = ({
   const [isLinkManagementAlertModalOpen, setIsLinkManagementAlertModalOpen] = useState(false);
   const [isLoadingLinkManagementEnabledChange, setIsLoadingLinkManagementEnabledChange] = useState(false);
   const [hasLinkManagementEnabledChangeError, setHasLinkManagementEnabledChangeError] = useState(false);
+  const formattedLinkExpirationDate = useMemo(() => moment.max(
+    [
+      ...couponsData.results.map(coupon => moment(coupon.endDate)),
```
Comment (macdiesel) [inline src/components/settings/SettingsAccessTab/SettingsAccessLinkManagement.jsx:49]: Nit: Could you leave a comment to our future selves with a bit more about the why we calculate it this way?

**Review by macdiesel — APPROVED** [review]
Looks good Long.

**Comment by macdiesel** [comment]
Fix the CI before you merge though...

File: src/components/settings/SettingsAccessTab/SettingsAccessLinkManagement.jsx [hunk 2]
```diff
@@ -178,12 +188,20 @@ const SettingsAccessLinkManagement = ({
...
+const mapDispatchToProps = dispatch => ({
+  updatePortalConfiguration: data => dispatch(updatePortalConfigurationEvent(data)),
 });
 
 SettingsAccessLinkManagement.propTypes = {
   enterpriseUUID: PropTypes.string.isRequired,
   isUniversalLinkEnabled: PropTypes.bool.isRequired,
-  dispatch: PropTypes.func.isRequired,
+  couponsData: PropTypes.shape({
+    results: PropTypes.array,
```
Comment (adamstankiewicz) [inline src/components/settings/SettingsAccessTab/SettingsAccessLinkManagement.jsx:202]: nit: I believe `PropTypes.array` is usually flagged by eslint, in favor of using `PropTypes.arrayOf(...)` here instead.

File: src/components/settings/SettingsContext.jsx [hunk 3]
```diff
@@ -28,10 +41,23 @@ function SettingsContextProvider({ children, enterpriseId }) {
 SettingsContextProvider.propTypes = {
   children: PropTypes.node.isRequired,
   enterpriseId: PropTypes.string.isRequired,
+  loadingCoupons: PropTypes.bool.isRequired,
+  couponsData: PropTypes.shape({
+    results: PropTypes.array,
```
Comment (adamstankiewicz) [inline src/components/settings/SettingsContext.jsx:46]: nit: `PropTypes.arrayOf(...)`

File: src/components/settings/SettingsContext.jsx [hunk 4]
```diff
@@ -1,23 +1,36 @@
...
   const {
     customerAgreement,
     loadingCustomerAgreement,
   } = useCustomerAgreementData({ enterpriseId });
 
+  useEffect(() => {
+    fetchCoupons();
+  }, []);
+
   const context = useMemo(() => ({
```
[the remark this answers is not in the document]
Comment (long74100) [reply src/components/settings/SettingsContext.jsx:24]: I think that makes sense

File: src/components/LoadingMessage/_LoadingMessage.scss [hunk 5]
```diff
@@ -7,7 +7,7 @@
   &.support,
   &.admin-register,
   &.user-activation,
-  &.subscriptions {
+  &.subscriptions, &.settings {
```
Comment (adamstankiewicz) [inline src/components/LoadingMessage/_LoadingMessage.scss:10]: nit: new line for `&.settings`

File: src/components/ProductTours/BrowseAndRequestTour.jsx [hunk 6]
```diff
@@ -12,19 +12,23 @@ import {
   BROWSE_AND_REQUEST_TOUR_COOKIE_NAME,
   TOUR_TARGETS,
 } from './constants';
+import { SubsidyRequestConfigurationContext } from '../subsidy-request-configuration';
 
 const cookies = new Cookies();
 
 const BrowseAndRequestTour = ({ enterpriseSlug, enableBrowseAndRequest }) => {
+  const { subsidyRequestConfiguration } = useContext(SubsidyRequestConfigurationContext);
+  const isEligibleForFeature = !!subsidyRequestConfiguration?.subsidyType;
```
Comment (adamstankiewicz) [inline src/components/ProductTours/BrowseAndRequestTour.jsx:21]: Should this also be adhering to `subsidyRequestConfiguration?.subsidyRequestsEnabled`?

File: src/components/ProductTours/tests/BrowseAndRequestTour.test.jsx [hunk 7]
```diff
@@ -45,27 +47,39 @@ const historyMock = (pathname = SUBSCRIPTION_PAGE_LOCATION) => ({
...
+const TourWithContext = ({ pathname = undefined, isEligibleForBrowseAndRequest = true }) => (
   <Router history={historyMock(pathname)}>
     <Provider store={store}>
-      <>
-        <BrowseAndRequestTour />
-        <p id={TOUR_TARGETS.SETTINGS_SIDEBAR}>Settings</p>
-      </>
+      <SubsidyRequestConfigurationContext.Provider value={{
+        subsidyRequestConfiguration: {
+          subsidyType: isEligibleForBrowseAndRequest ? SUPPORTED_SUBSIDY_TYPES.license : null,
```
Comment (adamstankiewicz) [inline src/components/ProductTours/tests/BrowseAndRequestTour.test.jsx:55]: Should this also include (and test for) `subsidyRequestsEnabled: true/false`?

File: src/components/ProductTours/BrowseAndRequestTour.jsx (hunk 6 above)
Comment (long74100) [reply src/components/ProductTours/BrowseAndRequestTour.jsx:21]: nope, subsidyRequestsEnabled defaults to false for all customers at the time of creation of the configuration. As long as the enterprise is eligible, the tour should be shown.

**Review by adamstankiewicz — APPROVED** [review]

**Pull request merged**